### PR TITLE
8309632: JDK 21 RDP1 L10n resource files update

### PR DESCRIPTION
**Pull request by justin-curtis-lu**
Please review this PR which updates the JDK's localized resources since the previous L10n translation drop (1/26).

To help with reviewing the changes, @jonathan-gibbons created a tool which displays the localized changes next to the original file's changes in UTF-8 native. Those files can be viewed [here](https://cr.openjdk.org/~jlu/output/) 

For example,

image

Please note that the HTML files only apply to .properties, and not .java resources.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8309632](https://bugs.openjdk.org/browse/JDK-8309632): JDK 21 RDP1 L10n resource files update (**Bug** - P3)

### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14430/head:pull/14430` \
`$ git checkout pull/14430`

Update a local copy of the PR: \
`$ git checkout pull/14430` \
`$ git pull https://git.openjdk.org/jdk.git pull/14430/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14430`

View PR using the GUI difftool: \
`$ git pr show -t 14430`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14430.diff">https://git.openjdk.org/jdk/pull/14430.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14430#issuecomment-1588172394)

### CODE REVIEW

**Comment by justin-curtis-lu** [comment]
@naotoj @wangweij @sormuras could you please review the changes that affect the languages you are familiar with when you have the chance. Please use the HTML files for this, as they will be in UTF-8 native and not unicode escaped sequences.

@prrace @jonathan-gibbons @pavelrappo @lahodaj @wangweij @igraves @AlanBateman @eirbjo could you please review the modules / changes you are familiar with, thank you.

File: src/jdk.jartool/share/classes/sun/tools/jar/resources/jar_de.properties [hunk 1]
```diff
@@ -104,14 +105,14 @@ main.usage.summary.try=Verwenden Sie "jar --help", um weitere Informationen anzu
...
 main.help.opt.any=\ In jedem Modus g\u00FCltige Vorgangsmodifikatoren:\n\n  -C DIR                     Zum angegebenen Verzeichnis wechseln und die folgende\n                             Datei aufnehmen
-main.help.opt.any.file=\  -f, --file=FILE            Der Name der Archivdatei. Wenn Sie dies auslassen, wird entweder stdin oder\n                             stdout verwendet, je nach Vorgang\n      --release VERSION      Speichert alle der folgenden Dateien in einem versionierten Verzeichnis\n                             der JAR-Datei (d.h. META-INF/versions/VERSION/)
+main.help.opt.any.file=\  -f, --file=FILE            Der Name der Archivdatei. Wenn Sie dies lauslassen, wird entweder stdin oder\n                             stdout verwendet, je nach Vorgang\n      --release VERSION      Speichert alle der folgenden Dateien in einem versionierten Verzeichnis\n                             der JAR-Datei (d.h. META-INF/versions/VERSION/)
```
Comment (eirbjo) [inline src/jdk.jartool/share/classes/sun/tools/jar/resources/jar_de.properties:115]: This change looks suspicious to me. The English word used is “omitted”. “Auslassen” is to “Leave out”, while “lauslassen” is to “let go”. A native German speaker should confirm, I have only high school level German..

File: src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler_ja.properties [hunk 2]
```diff
@@ -927,6 +953,18 @@ compiler.err.unclosed.str.lit=\u6587\u5B57\u5217\u30EA\u30C6\u30E9\u30EB\u304C\u
 
 compiler.err.unclosed.text.block=\u9589\u3058\u3089\u308C\u3066\u3044\u306A\u3044\u30C6\u30AD\u30B9\u30C8\u30FB\u30D6\u30ED\u30C3\u30AF
 
+compiler.err.string.template.is.not.well.formed=\u6587\u5B57\u5217\u30C6\u30F3\u30D7\u30EC\u30FC\u30C8\u304C\u6574\u5F62\u5F0F\u3067\u306F\u3042\u308A\u307E\u305B\u3093
+
+compiler.err.text.block.template.is.not.well.formed=\u30C6\u30AD\u30B9\u30C8\u30FB\u30D6\u30ED\u30C3\u30AF\u30FB\u30C6\u30F3\u30D7\u30EC\u30FC\u30C8\u304C\u6574\u5F62\u5F0F\u3067\u306F\u3042\u308A\u307E\u305B\u3093
+
```
Comment (naotoj) [inline src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler_ja.properties:959]: The translation of "well-formed" is incorrect. It is not 整数式

File: src/java.desktop/macosx/classes/com/apple/laf/resources/aqua_ja.properties [hunk 3]
```diff
@@ -60,6 +60,7 @@ FileChooser.saveTitle.textAndMnemonic=\u4FDD\u5B58
 FileChooser.openTitle.textAndMnemonic=\u958B\u304F
 FileChooser.newFolderExistsError.textAndMnemonic=\u305D\u306E\u540D\u524D\u306F\u3059\u3067\u306B\u4F7F\u7528\u3055\u308C\u3066\u3044\u307E\u3059
 FileChooser.chooseButton.textAndMnemonic=\u9078\u629E
+FileChooser.chooseButtonToolTip.textAndMnemonic=\u9078\u629E\u3057\u305F\u30D5\u30A1\u30A4\u30EB\u3092\u8FFD\u52A0\u3057\u307E\u3059
```
Comment (naotoj) [inline src/java.desktop/macosx/classes/com/apple/laf/resources/aqua_ja.properties:63]: The translation seems wrong. It means to *add* the selected file, which is not the same meaning as in English file

File: src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler_ja.properties [hunk 4]
```diff
@@ -477,6 +499,10 @@ compiler.err.not.in.profile={0}\u306F\u30D7\u30ED\u30D5\u30A1\u30A4\u30EB''{1}''
 # 0: symbol
 compiler.warn.forward.ref=\u521D\u671F\u5316\u3055\u308C\u308B\u524D\u306E\u5909\u6570''{0}''\u3092\u53C2\u7167\u3057\u3088\u3046\u3068\u3057\u307E\u3057\u305F
 
+compiler.warn.possible.this.escape=\u30B5\u30D6\u30AF\u30E9\u30B9\u524D\u306B\u6307\u5B9A\u53EF\u80FD\u306A''this''\u30A8\u30B9\u30B1\u30FC\u30D7\u306F\u5B8C\u5168\u306B\u521D\u671F\u5316\u3055\u308C\u307E\u3059
+
+compiler.warn.possible.this.escape.location=\u4E8B\u524D\u306B\u6307\u5B9A\u53EF\u80FD\u306A''this''\u30A8\u30B9\u30B1\u30FC\u30D7\u306F\u547C\u51FA\u3057\u3092\u4ECB\u3057\u3066\u3053\u3053\u306B\u51FA\u73FE\u3057\u307E\u3059
+
```
Comment (naotoj) [inline src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler_ja.properties:505]: These translations are very cryptic. Cannot understand it at first hand.

**Review by naotoj — COMMENTED** [review]
Left some comments on the translations mainly in Japanese. It is now very easy to look at the l10n changes in the generated HTML. One small comment to the tool is that it would be nice if the order in HTML (alphabetically sorted currently) aligns with the order in the actual resource file. It is a great improvement anyways.

File: src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler_de.properties [hunk 5]
```diff
@@ -136,6 +136,9 @@ compiler.err.annotation.value.not.allowable.type=Annotationswert hat keinen zul\
 
 compiler.err.expression.not.allowable.as.annotation.value=Ausdruck ist als Annotationswert nicht zul\u00E4ssig
 
+# 0: type, 1: name
+compiler.err.annotation.unrecognized.attribute.name=Annotation @{0} weist das unbekannte Attribut "{1}" auf
```
[the remark this answers is not in the document]
Comment (justin-curtis-lu) [reply src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler_de.properties:140]: I will consult the language specialist on this. I remember in previous discussions that certain languages would format punctuation differently, but let ask about this specific instance.

File: src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler_ja.properties (hunk 2 above)
Comment (justin-curtis-lu) [reply src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler_ja.properties:959]: For translations that are incorrect, I can file a bug against the tool. Such a fix would probably not happen soon, can you suggest an alternative for now?

File: src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler_ja.properties [hunk 6]
```diff
@@ -477,6 +499,10 @@ compiler.err.not.in.profile={0}\u306F\u30D7\u30ED\u30D5\u30A1\u30A4\u30EB''{1}''
 # 0: symbol
 compiler.warn.forward.ref=\u521D\u671F\u5316\u3055\u308C\u308B\u524D\u306E\u5909\u6570''{0}''\u3092\u53C2\u7167\u3057\u3088\u3046\u3068\u3057\u307E\u3057\u305F
 
+compiler.warn.possible.this.escape=\u30B5\u30D6\u30AF\u30E9\u30B9\u524D\u306B\u6307\u5B9A\u53EF\u80FD\u306A''this''\u30A8\u30B9\u30B1\u30FC\u30D7\u306F\u5B8C\u5168\u306B\u521D\u671F\u5316\u3055\u308C\u307E\u3059
```
Comment (naotoj) [inline src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler_ja.properties:502]:
```suggestion
compiler.warn.possible.this.escape=\u30b5\u30d6\u30af\u30e9\u30b9\u304c\u521d\u671f\u5316\u3055\u308c\u308b\u524d\u306e''this''\u30a8\u30b9\u30b1\u30fc\u30d7\u306e\u53ef\u80fd\u6027\u304c\u3042\u308a\u307e\u3059
```

File: src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler_ja.properties [hunk 7]
```diff
@@ -477,6 +499,10 @@ compiler.err.not.in.profile={0}\u306F\u30D7\u30ED\u30D5\u30A1\u30A4\u30EB''{1}''
...
+
+compiler.warn.possible.this.escape.location=\u4E8B\u524D\u306B\u6307\u5B9A\u53EF\u80FD\u306A''this''\u30A8\u30B9\u30B1\u30FC\u30D7\u306F\u547C\u51FA\u3057\u3092\u4ECB\u3057\u3066\u3053\u3053\u306B\u51FA\u73FE\u3057\u307E\u3059
```
Comment (naotoj) [inline src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler_ja.properties:504]:
```suggestion
compiler.warn.possible.this.escape.location=\u4e8b\u524d\u306e''this''\u30a8\u30b9\u30b1\u30fc\u30d7\u306f\u3053\u306e\u547c\u3073\u51fa\u3057\u3067\u51fa\u73fe\u3057\u307e\u3059
```

File: src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler_ja.properties [hunk 8]
```diff
@@ -927,6 +953,18 @@ compiler.err.unclosed.str.lit=\u6587\u5B57\u5217\u30EA\u30C6\u30E9\u30EB\u304C\u
 
 compiler.err.unclosed.text.block=\u9589\u3058\u3089\u308C\u3066\u3044\u306A\u3044\u30C6\u30AD\u30B9\u30C8\u30FB\u30D6\u30ED\u30C3\u30AF
 
+compiler.err.string.template.is.not.well.formed=\u6587\u5B57\u5217\u30C6\u30F3\u30D7\u30EC\u30FC\u30C8\u304C\u6574\u5F62\u5F0F\u3067\u306F\u3042\u308A\u307E\u305B\u3093
```
Comment (naotoj) [inline src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler_ja.properties:956]:
```suggestion
compiler.err.string.template.is.not.well.formed=\u6587\u5b57\u5217\u30c6\u30f3\u30d7\u30ec\u30fc\u30c8\u304c\u9069\u683c\u3067\u306f\u3042\u308a\u307e\u305b\u3093
```

File: src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler_ja.properties [hunk 9]
```diff
@@ -927,6 +953,18 @@ compiler.err.unclosed.str.lit=\u6587\u5B57\u5217\u30EA\u30C6\u30E9\u30EB\u304C\u
...
+
+compiler.err.text.block.template.is.not.well.formed=\u30C6\u30AD\u30B9\u30C8\u30FB\u30D6\u30ED\u30C3\u30AF\u30FB\u30C6\u30F3\u30D7\u30EC\u30FC\u30C8\u304C\u6574\u5F62\u5F0F\u3067\u306F\u3042\u308A\u307E\u305B\u3093
```
Comment (naotoj) [inline src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler_ja.properties:958]:
```suggestion
compiler.err.text.block.template.is.not.well.formed=\u30c6\u30ad\u30b9\u30c8\u30fb\u30d6\u30ed\u30c3\u30af\u30fb\u30c6\u30f3\u30d7\u30ec\u30fc\u30c8\u304c\u9069\u683c\u3067\u306f\u3042\u308a\u307e\u305b\u3093
```

**Review by naotoj — CHANGES_REQUESTED** [review]
Left some suggestions. Should be looked at by translators later.

File: src/java.desktop/macosx/classes/com/apple/laf/resources/aqua_ja.properties (hunk 3 above)
[the remark this answers is not in the document]
Comment (justin-curtis-lu) [reply src/java.desktop/macosx/classes/com/apple/laf/resources/aqua_ja.properties]: Thank you Naoto, all your recommended translations have been updated. The HTML diffs have also been updated as well.

**Comment by justin-curtis-lu** [comment]
Thank you for the review @wangweij,

I have addressed these recommendations

>     * `src/jdk.compiler/share/classes/com/sun/tools/javac/resources/javac`:
> 
> 
> `javac.opt.Xlint.desc.output-file-clash`: "native header files" is translated into "本机标头文件". The "本机" is strange, I'd rather just say "头文件" as there is no non-native header file.
> 
>     * `src/jdk.jdeps/share/classes/com/sun/tools/javap/resources/javap`
> 
> 
> I don't think the new translation is better than the old one.

For the other comments, I will bring them to the language specialist for updated translations. Ideally we can get those in for this translation drop, but if not, they can go in during the second translation drop during RDP2. 

I am also hesitant to make too many manual changes, as they will just get overwritten by the tool in the future, so it's best to get it fixed in the tool first.

**Review by naotoj — APPROVED** [review]
Looks good, Justin

File: src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler_de.properties (hunk 5 above)
[the remark this answers is not in the document]
Comment (justin-curtis-lu) [reply src/jdk.compiler/share/classes/com/sun/tools/javac/resources/compiler_de.properties:140]: The language specialist confirmed the double quotes are locale specific for de.

**Comment by jonathan-gibbons** [comment]
> Left some comments on the translations mainly in Japanese. It is now very easy to look at the l10n changes in the generated HTML. One small comment to the tool is that it would be nice if the order in HTML (alphabetically sorted currently) aligns with the order in the actual resource file. It is a great improvement anyways.

1. It is not easy to recover the order in the actual resource file: the contents are read into a `Properties` object which uses a `Hashtable` to store the data. 
2. There is no guarantee that the order is the same in all the resource files, so we would have to choose one, presumably the default one, to determine the presentation order.

**Comment by naotoj** [comment]
Makes sense. Thanks for the explanation @jonathan-gibbons

**Pull request closed without merging**